### ISSUE TITLE
Require velero s3 secret name input in either ramen config map or VRG - sprint 6

### ISSUE THREAD

**Issue by hatfieldbrian** [RamenDR/ShioRamen#48]
I suggest this input be a field in the s3 profile (ramen config map).  Maybe pass as input to velero protect and recover routines to preserve velero-driver interface.
Since ramen config is not an API server object, I don't think it can be required, however, the controller can check for it and set a status condition
I think we just need some documentation to the user, maybe inline in the `ramenconfig_types.go` for now.

**Comment by tjanssen3** [comment]
This is a code snippet (edited for conciseness) from `kubeobjects.go` that uses the S3 Secret info:
```go
return &velero.BackupStorageLocation{
		TypeMeta: metav1.TypeMeta{
			APIVersion: "velero.io/v1",
			Kind:       "BackupStorageLocation",
		},
		Spec: velero.BackupStorageLocationSpec{
			Provider: "aws",
			StorageType: velero.StorageType{
				ObjectStorage: &velero.ObjectStorageLocation{
					Bucket: bucketName,
					Prefix: s3KeyPrefix + KubeObjectsPath,
				},
			},
                        // this is the important section:
			Credential: &corev1.SecretKeySelector{
				LocalObjectReference: corev1.LocalObjectReference{
					Name: secretName,
				},
				Key: secretKeyName,
			},
		},
}
```

The `Credential` field is a `SecretKeySelector`, which requires a `LocalObjectReference` field that takes the name of the Secret. Unfortunately, `LocalObjectReference` only operates in the host namespace of the object - which for the VRG is the namespace it's protecting and probably not the Velero namespace. This means that even if we get the user to pass in s3 Secret name and namespace to the Ramen ConfigMap, the VRG code can't effectively use that Secret, unless I'm missing something. 

@hatfieldbrian  Would an acceptable alternate solution be to document the S3 Secret requirements for Velero so users know they need this?

**Comment by hatfieldbrian** [comment]
For MVP, I think the only thing needed from the config map is the name of the secret.  The namespace is already an input.  VRG reconciler just passes it through to the velero-driver and velero accesses it.

**Comment by tjanssen3** [comment]
I thought I'd follow up here because I was a little confused about this request. Originally, I thought we were using this to access a Secret across a namespace, which I don't believe you can do (`LocalObjectReference` only checks the namespace it's in). However, the part of the Ramen code that requires Credentials is in the Velero namespace already, so there's no cross-namespace lookup happening. To this end, I added the requested changes on PR #61. 

One thing we could improve in the future is the key-lookup mechanism, but I think we can omit that and document it for MVP. Right now we need users to use key `"aws"` in the Velero S3 Secret and specify login/password within that struct. Later on, we could change that name, but I think it's ok to leave it as-is for now and document that requirement.

**Comment by hatfieldbrian** [comment]
Fixed in #61
